cmd/gateway/main.go

```diff
@@ -4,10 +4,212 @@
 package main
 
 import (
-	"storj.io/storj/cmd/uplink/cmd"
+	"context"
+	"crypto/rand"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+
+	base58 "github.com/jbenet/go-base58"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/fpath"
+	"storj.io/storj/pkg/cfgstruct"
+	"storj.io/storj/pkg/miniogw"
 	"storj.io/storj/pkg/process"
+	"storj.io/storj/pkg/storage/streams"
+	"storj.io/storj/pkg/storj"
+)
+
+// GatewayFlags configuration flags
+type GatewayFlags struct {
+	APIKey             string `default:"" help:"the api key to use for the satellite" setup:"true"`
+	GenerateMinioCerts bool   `default:"false" help:"generate sample TLS certs for Minio GW" setup:"true"`
```

Comment:
Maybe make the flag `--generate-test-certs`.

Reply:
sounds ok, done